pkg/virt-api/rest/subresource.go

```diff
@@ -560,6 +561,13 @@ func (app *SubresourceAPIApp) RenameVMRequestHandler(request *restful.Request, r
 		return
 	}
 
+	nameErrs := k8sapivalidation.NameIsDNSSubdomain(opts.NewName, false)
+	if len(nameErrs) > 0 {
+		message := fmt.Sprintf("The VM's new name is not valid: %s", strings.Join(nameErrs, "; "))
```

Comment:
We should put this validation check server side in the `pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go` instead.

That will still cause the request to rename using virtctl to fail, but it will be validated on the server side with the rest of the spec/status validators instead.  So, the request to PATCH will fail here in the client if the validation webhook rejects the change.

Reply:
Done.